src/Controller/FrontPageConfigurationController.php

```diff
@@ -20,37 +22,87 @@
   * @IsGranted("ROLE_ADMIN")
   */
 final class FrontPageConfigurationController extends AbstractController {
-    public function frontPageConfig(ForumRepository $fr, SubmissionRepository $sr, ForumConfigurationRepository $fcr, EntityManager $em, Request $request) {
-        $data = new ForumConfigurationData($fcr->findSitewide());
 
-        $form = $this->createForm(ForumConfigurationType::class, $data);
+    public function frontPageConfig(
+        ForumRepository $fr,
+        SubmissionRepository $sr,
+        ForumConfigurationRepository $fcr,
+        UserRepository $ur,
+        EntityManager $em,
+        Request $request
+    ) {
+        $announcementForumName = "announcements";
+        $data = new NewForumAnnouncementData();
+
+        $form = $this->createForm(NewForumAnnouncementType::class, $data);
         $form->handleRequest($request);
 
         $message = "";
+        $messageClass = "";
 
         if ($form->isSubmitted() && $form->isValid()) {
-            $fc = $data->toForumConfiguration();
+            // Find the user and forum for submission.
+            $user = $ur->loadUserByUsername($this->get('security.token_storage')->getToken()->getUser()->getUsername());
+            $forum = $fr->findOneByCaseInsensitiveName($announcementForumName);
+
+            if($forum != null) {
+                // Create the submission.
+                $submission = new Submission(
+                    $data->threadTitle,
+                    null,
+                    $data->threadContent,
+                    $forum,
+                    $user,
+                    null
+                );
+                $em->persist($submission);
+                $em->flush();
 
-            // For sitewide settings, we need to force this to null.
-            $fc->setForumId(null);
+                // Load the current announcement's row id. Sitewide announcemnts have null forum id.
+                $fc = $fcr->findSitewide();
```

Comment:
Is there ever a situation this could return null?
On first migration, before the first announcement has been made?

Reply:
Yes, the only reason this would ever be null is if the announcement forum hadn't been made yet (or deleted for whatever reason).